server-spi-private/src/main/java/org/keycloak/models/AccountRoles.java

```diff
@@ -24,8 +24,9 @@
 
     String VIEW_PROFILE = "view-profile";
     String MANAGE_ACCOUNT = "manage-account";
+    String INITIATE_ACTION = "initiate-action";
     String MANAGE_ACCOUNT_LINKS = "manage-account-links";
 
-    String[] ALL = {VIEW_PROFILE, MANAGE_ACCOUNT};
+    String[] ALL = {VIEW_PROFILE, MANAGE_ACCOUNT, INITIATE_ACTION};
```

Comment:
So did I misunderstand what you wrote in the [JIRA](https://issues.jboss.org/browse/KEYCLOAK-9366) under acceptance criteria?  I took this to imply that we need a new role.  It says,
> It should be possible to configure which applications is permitted to initiate application initiated actions. Most likely through a role like it is done for view-account and manage-account.

Reply:
All three of the above bullet points are now addressed.

- OIDC error and error_description are returned.  There is also a new "silent_cancel" feature that allows the application to request no error on cancel.  This is important for apps that display the URL in the address bar. 
- Cancel was already permitted only for AIA.
- Removed additional role check.  It now uses the manage-account role instead.

Also, I did not squash the commits.  I want to keep both commits around because I may need to look back.